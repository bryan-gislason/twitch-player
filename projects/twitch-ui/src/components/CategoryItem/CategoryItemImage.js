import { useTheme } from "@twitch-player/themes/dist/hooks";
import PropTypes from "prop-types";
import React, { memo } from "react";
import { Image, StyleSheet } from "react-native";
import { processImageTemplate } from "../../utils";

const CategoryItemImage = ({
  url,
  name,
  width: boxArtWidth,
  height: boxArtHeight,
  testID = "category-item-image",
}) => {
  const [{ colors, layout }] = useTheme();

  const imageURL = processImageTemplate(
    url,
    layout.maxCategoryWidth,
    layout.maxCategoryHeight
  );

  const imageStyle = [
    styles.image,
    {
      backgroundColor: colors.itemBackground,
      height: boxArtHeight,
      width: boxArtWidth,
    },
  ];

  return (
    <Image
      source={{ uri: imageURL }}
      accessibilityLabel={name}
      style={imageStyle}
      testID={testID}
    />
  );
};

CategoryItemImage.propTypes = {
  url: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  width: PropTypes.number.isRequired,
  height: PropTypes.number.isRequired,
  testID: PropTypes.string,
};

const styles = StyleSheet.create({
  image: {
    display: "block",
  },
});

export default memo(CategoryItemImage);
