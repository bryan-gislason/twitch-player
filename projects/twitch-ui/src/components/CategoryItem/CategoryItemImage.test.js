import React from "react";
import { render } from "../../testing";
import CategoryItemImage from "./CategoryItemImage";

describe("CategoryItemImage", () => {
  let instance;

  beforeEach(() => {
    instance = render(
      <CategoryItemImage
        url="categories/Fortnite-376x500.jpg"
        name="Fortnite"
        width={188}
        height={250}
      />
    );
  });

  it("renders a category item image", () => {
    expect(instance.getByTestId("category-item-image")).toMatchSnapshot();
  });
});
