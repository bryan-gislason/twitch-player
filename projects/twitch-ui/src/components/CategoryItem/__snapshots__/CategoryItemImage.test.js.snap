// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryItemImage renders a category item image 1`] = `
<div
  aria-label="Fortnite"
  class="rn-alignItems-1oszu61 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-flexBasis-1mlwlqe rn-flexDirection-eqz5dr rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-overflowX-11yh6sk rn-overflowY-buy8e9 rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-zIndex-1lgpqti"
  data-testid="category-item-image"
  style="background-color: rgb(218, 216, 222); display: block; height: 250px; width: 188px;"
>
  <div
    class="rn-alignItems-1oszu61 rn-backgroundColor-1niwhzg rn-backgroundPosition-vvn4in rn-backgroundRepeat-u6sd8q rn-backgroundSize-4gszlv rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-bottom-1p0dtai rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1qe8dj5 rn-flexBasis-1mlwlqe rn-flexDirection-eqz5dr rn-height-1pi2tsx rn-left-1d2f490 rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-u8s1d rn-right-zchlnj rn-top-ipm5af rn-width-13qz1uu rn-zIndex-1wyyakw"
    style="background-image: url(categories/Fortnite-376x500.jpg);"
  />
  <img
    alt="Fortnite"
    class="rn-bottom-1p0dtai rn-height-1pi2tsx rn-left-1d2f490 rn-opacity-1272l3b rn-position-u8s1d rn-right-zchlnj rn-top-ipm5af rn-width-13qz1uu rn-zIndex-1wyyakw"
    draggable="false"
    src="categories/Fortnite-376x500.jpg"
  />
</div>
`;
